src/modules/AutoinstConfig.rb

```diff
@@ -214,114 +219,157 @@ def setProposalList(l)
           end
-          result = URL.Parse(cmdLine)
-          @OriginalURI = cmdLine
         end
+
+        dlg = Left(ComboBox(Id(:choose), _("Choose Profile"), dummy))
+
+        UI.OpenDialog(VBox(dlg, PushButton(Id(:ok), Label.OKButton)))
+        UI.UserInput
+
+        profile_location = Ops.get(
+          comment2url,
+          Convert.to_string(UI.QueryWidget(Id(:choose), :Value)),
+          ""
+        )
+
+        UI.CloseDialog
+
+      # just one provider
+      elsif Builtins.size(slpData) == 1
+        profile_location = Builtins.substring(
+          Ops.get_string(slpData, [0, "srvurl"], ""),
+          17
+        )
+
+      # Nothing returned by SLP query
+      else
+        log.error "slp query for 'autoyast' failed"
+        Report.Error(_("No 'autoyast' provider has been found via SLP."))
       end
 
+      profile_location
+    end
+
+    # Updates or extends the profile location according to defaults
+    # @param [String] AutoYast profile location as defined on commandline
+    # @return [String] updated profile location
+    def update_profile_location(profile_location)
+      profile_location = deep_copy(profile_location)
 
-      if Ops.get_string(result, "scheme", "") == ""
+      if profile_location.nil? || profile_location == ""
+        # FIXME: reevaluate this statement
+        #
         # Autoinstall mode was not activated from command line.
         # There must be a floppy with an 'autoinst.xml' in order
         # to be able to reach this point, so we set floppy with
         # autoinst.xml as the control file.
-
-        result = Builtins.add(result, "scheme", "floppy")
-        result = Builtins.add(result, "path", "/autoinst.xml")
+        profile_location = "floppy:///#{DEFAULT_PROFILE_NAME}"
+      elsif profile_location == "default"
+        profile_location = "file:///#{DEFAULT_PROFILE_NAME}"
+      # bsc#987858: autoyast=usb checks for the default profile
+      elsif profile_location == "usb"
+        profile_location = "usb:///#{DEFAULT_PROFILE_NAME}"
+      elsif profile_location == "slp"
+        profile_location = find_slp_autoyast
+        return if profile_location.nil?
       end
-      @urltok = deep_copy(result)
 
-      @scheme = Ops.get_string(@urltok, "scheme", "default")
-      @host = Ops.get_string(@urltok, "host", "")
-      @filepath = Ops.get_string(@urltok, "path", "")
-      @port = Ops.get_string(@urltok, "port", "")
-      @user = Ops.get_string(@urltok, "user", "")
-      @pass = Ops.get_string(@urltok, "pass", "")
+      profile_location
+    end
+
+    # Processes location of the profile given as a parameter.
+    # @param [String] AutoYast profile location as defined on commandline
+    # @example autoyast=http://www.server.com/profiles/
+    # Fills internal variables
+    def ParseCmdLine(profile_location)
+
+      log.info "AutoYast profile location #{profile_location}"
+
+      profile_location = update_profile_location(profile_location)
```

Comment:
You could rename the variables to something like:

```ruby
profile_location = canonical_location(proposed_location)
```

Well, I suck at naming, but you get the idea.